build/xtask/src/main.rs

```diff
@@ -121,11 +120,28 @@ enum Xtask {
     },
 }
 
+#[derive(Clone, Debug, Parser)]
+pub struct HumilityArgs {
+    /// Path to the image configuration file, in TOML.
+    cfg: PathBuf,
+
+    /// Request verbosity from tools we shell out to.
+    #[clap(short, long)]
+    verbose: bool,
+
+    /// Extra options to pass to Humility
+    #[clap(last = true)]
+    extra_options: Vec<String>,
+}
+
 // For commands which may execute on specific packages, this enum
 // identifies the set of packages that should be operated upon.
 fn main() -> Result<()> {
     let xtask = Xtask::parse();
+    run(xtask)
+}
 
+fn run(xtask: Xtask) -> Result<()> {
```

Comment:
what's the reason to extract everything but arg parsing into a separate function?

Reply:
So I can re-call into `run` [further down](https://github.com/oxidecomputer/hubris/blob/f194910e81ae34e1ac21a07b8a48ddf78697ff78/build/xtask/src/main.rs#L187)